### ISSUE TITLE
Allow multiple sites to be hosted from one Cloudfront 

### ISSUE THREAD

**Issue by jonahallibone** [serverless-nextjs/serverless-next.js#1015]
**Is your feature request related to a problem? Please describe.**
I am not sure if this is supported at the moment, but I have two project I'd like to host under a single domain. My company has a marketing site with our landing pages, and another site which contains our web app. Our web app contains a public view information on companies, which then requires authentication to view more detailed and specific information.

**Describe the solution you'd like**
 Currently I have two next JS projects for `marketing-site` and `web-app`. There are no routing conflicts between the two. If I set the CF distro id to be the same on both projects, created different S3 buckets, and named  the lambdas differently, I'd like to be able to deploy the site the CF without destroying the resources for the other site.

**Describe alternatives you've considered**
The alternative would be to have a site with our marketing on the main domain, and the application on the subdomain. For reasons related to SEO and ease of use, we'd like to avoid this.

**Comment by dphang** [comment]
Yeah I think this has been brought up before, though I think it is not just a small task because right now the handler uses manifests from just a single Next.js app. We are working to refactor the core handler code to be more maintainable/generic, I think after that we can consider it. Definitely feel free to post some thoughts on how you think the design should be like.

**Comment by jonahallibone** [comment]
@dphang Will do. I'll take a look at how the current handler is set up and whether or not I feel qualified to make a suggestion haha

**Comment by davegariepy** [comment]
@jonahallibone this sounds like a use case for [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites)

I don't think it is implemented yet in this package though.
